File: src/main/java/model/Card.java
package model;

public record Card(String rank, String suit) {
    public int value() {
        return switch (rank) {
            case "Ace" -> 11;
            case "King", "Queen", "Jack" -> 10;
            default -> Integer.parseInt(rank);
        };
    }

    @Override
    public String toString() {
        return String.format("\"%s of %s\"", rank, suit);
    }
}

File: Blackjack/Models/Card.cs
namespace Blackjack.Models;

public record Card(Rank Rank, Suit Suit)
{
    public int Value => Rank switch
    {
        Rank.ACE => 11,
        Rank.KING or Rank.QUEEN or Rank.JACK => 10,
        _ => (int)Rank
    };

    public string DisplayName => $"{Rank} of {Suit}";

    public override string ToString()
    {
        return $"Card({Rank}, {Suit})";
    }
}
